### ISSUE TITLE
Unexpected Behavior (maybe?) for Internal Energy and Liquid Water as a function of pressure

### ISSUE THREAD

**Issue by dansteingart** [CoolProp/CoolProp#1241]
The standard treatment single phase incompressible fluids (e.g. water) is that the internal energy should only be a function of pressure, while the enthalpy may change as

H = U + pV

So, I am wondering why in coolprop 6, when I examine the internal energy of water for a constant temperature 

```python
from CoolProp.CoolProp import PropsSI as pr

p1 = 1e5
p2 = 1e6
T = 300
u1 = pr('U','T',T,'P',p1,'water')
u2 = pr('U','T',T,'P',p2,'water')

print 'u1 =',u1,'J/kg'
print 'u2 =',u2,'J/kg'

h1 = pr('H','T',T,'P',p1,'water')
h2 = pr('H','T',T,'P',p2,'water')
print
print 'h1 =',h1,'J/kg'
print 'h2 =',h2,'J/kg'

v1 = 1/pr('D','T',T,'P',p1,'water')
v2 = 1/pr('D','T',T,'P',p2,'water')

print
print 'v1 =',v1,'m^3/kg'
print 'v2 =',v2,'m^3/kg'

print
print 'u1 + p1*v1 =',u1+p1*v1,'J/kg'
print 'u2 + p2*v2 =',u2+p2*v2,'J/kg'
```
yields
```
p1 = 100000.0 Pa
p2 = 1000000.0 Pa
T = 300 K

u1 = 112553.334133 J/kg
u2 = 112478.998245 J/kg

h1 = 112653.679689 J/kg
h2 = 113482.047493 J/kg

v1 = 0.00100345555938 m^3/kg
v2 = 0.00100304924695 m^3/kg

u1 + p1*v1 = 112653.679689 J/kg
u2 + p2*v2 = 113482.047492 J/kg
```
CoolProp shows a slight, but not insignificant _decrease_ in the internal energy as a function of increase pressure.

Generally I have understood that for an incompressible fluid there is no change in internal energy with pressure at constant temperature, this is heart of the incompressible assumption.   But if there were to be a change, the trend ought to be that internal energy should increase with pressure otherwise there would be a driving force for water to seek higher pressure rather than lower pressure.  

Is this the expected behavior for the equation of state model used in the calculation?  If so can someone provide citation in which this trend is explained and even better some prose that provides physical intuition?

Thanks in advance.

**Comment by dansteingart** [comment]
To give myself another gut check, entropy seems to behave correctly

```
s1 = 393.062433815 J/(kg K)
s2 = 392.813902894 J/(kg K)
```

**Comment by dansteingart** [comment]
FWIW, steam97 shows the same behavior (u decreases as p increases for fixed T),  http://www.steamtablesonline.com/steam97web.aspx, so I'm guessing it has to do with the EoS formulation.  I guess what remains then is: is this physical reality or an artifact of fit.  

![image](https://cloud.githubusercontent.com/assets/152047/18811274/c4ac932c-8278-11e6-8be8-7c17a786b917.png)

As I understand this it should be the latter, but I would appreciate some confirmation.  Thanks in advance.

**Comment by dansteingart** [comment]
To beat a dead horse, one can do the above with the IAPWS in python and not have to pay any money

```python
from iapws import IAPWS97

s1 = IAPWS97(P=.1, T=300)
s2 = IAPWS97(P=1, T=300)

print "U(T=300 K,P=1e5 Pa) =",s1.u,"kJ/kg"
print "U(T=300 K,P=1e6 Pa) =",s2.u,"kJ/kg"
```

which is gives the same answer as CoolProp

```
U(T=300 K,P=1e5 Pa) = 112.563477841 kJ/kg
U(T=300 K,P=1e6 Pa) = 112.489253129 kJ/kg
```

So then I RFTM and saw that yes, this is that the basis of the calculation for CoolProp's calculation.  [Wagner 2002](http://scitation.aip.org/content/aip/journal/jpcrd/31/2/10.1063/1.1461829) 

From the LiquidWater reference pdf for IAPWS (http://www.iapws.org/relguide/LiquidWater.pdf).  

![image](https://cloud.githubusercontent.com/assets/152047/18811990/27ad58b2-8292-11e6-90e4-a3af8d833271.png)

Perhaps it comes from a slight difference in the _vdP_ term used to estimate the difference between _g_  and _g0_ in the first equation and the pv correction applied when then backing out _u_.  A quick read of the original source does not indicate the behavior shown above, there is just a brief line that reads

> In the liquid – water region, small changes in density along an isotherm cause large changes in pressure. For this reason, due to an accumulation of small errors, a particular computer code may fail to return the zeros in Eq.  6.7  for the saturated liquid at the triple-point temperature. In order to avoid this, it is advisable to readjust the constants n1 and n2 in Eq.  6.5  by imposing the condition u  = 0, s = 0 with the desired accuracy.

This is the only specific mention of internal energy and liquid water, so I am guessing this may be an artifact of the imposed solution above.  If anyone has any further clarity on this it would be most appreciated.  I'll leave this open for now.

**Comment by ibell** [comment]
Here are some things to ponder..
1) Water is not actually incompressible.  Ok, its not very compressible at all, but it is not completely incompressible.  You can see this by calculating d(rho)/dp|T, which for a truly incompressible fluid is zero.  But CoolProp shows you it isn't quite zero:
```
In [4]: CP.PropsSI('d(Dmolar)/d(P)|T','T',300,'P',101325,'Water')
Out[4]: 2.492122910745039e-05
```

2) CoolProp implements the IAPWS-95 scientific formulation for water (http://www.coolprop.org/fluid_properties/fluids/Water.html#fluid-water), just like all these other packages (REFPROP does too).  The numbers should come out to within a few ppb (http://www.coolprop.org/fluid_properties/fluids/Water.html#refprop-validation-data) of other properly implemented libraries.  The IAPWS IF97 formulation for industrial use is also implemented, but doesn't give as comprehensive (or accurate) coverage as IAPWS-95.

In summary, I don't think there is anything wrong with CoolProp, I think your assumption of true incompressibility is the basis of your questions.

**Comment by dansteingart** [comment]
@ibell thanks for your reply and I apologize for not looking deeper into the backbone of CoolProp before my first post (I figured in out in my second self reply)

And with respect to the incompressible assumption I agree with you completely, buried in my initial post:

>But if there were to be a change, the trend ought to be that internal energy should increase with pressure otherwise there would be a driving force for water to seek higher pressure rather than lower pressure.

To clarify my  question: if the incompressible assumption isn't constrained, what is the physical basis of the internal energy decreasing?

So [I understand] (https://github.com/CoolProp/CoolProp/issues/1241#issuecomment-249395733) that CoolProp implements IAPWS-95, and that this EOS model leads to the numerical behavior, but I am trying to understand if this is physical or a computational artifiact.  

I cannot parse this from the [source material](http://scitation.aip.org/content/aip/journal/jpcrd/31/2/10.1063/1.1461829), so I'm wondering if anyone has clean explanation, other than the math says so, of why ∆U decreases as ∆P increases for liquid water under an isothermal process?

**Comment by ibell** [comment]
Hmm this is a good brainteaser question.  Clearly IAPWS says du/dp|T is negative:
```
In [5]: CP.PropsSI('d(Umolar)/d(P)|T','T',300,'P',101325,'Water')
Out[5]: -1.4895120100181738e-06
```
but I don't have a good intuitive understanding of what that means..  I can ask my colleagues on Monday.

**Comment by dansteingart** [comment]
Thanks!  Also you taught me a couple of CoolProp tricks, so double thanks!

**Comment by dansteingart** [comment]
I think I have a better understanding now, also thanks to CoolProp.  When I plot:

![image](https://cloud.githubusercontent.com/assets/152047/18812894/7b0b5526-82b1-11e6-9aff-fe70943b061a.png)

and include the phase change, the internal energy ought to decrease as the phase condenses and continues to localize: I wouldn't expect the internal energy of a liquid to be higher than that of a vapor at given temperature.  My challenge is that I was focussed on the work going _into_ the system and not how that work was affecting the state of the system.  

But since Q - W = ∆U, then Q = W + ∆U, and W is negative (work in) and U is negative (IAPWS), so heat is negative, therefore leaving the system. I was neglecting the heat rejection required to execute this process. 

@ibell thanks for your help!
